src/onnx/parse_shrink.cpp

```diff
@@ -0,0 +1,102 @@
+
+    instruction_ref parse(const op_desc&,
+                          const onnx_parser& parser,
+                          const onnx_parser::node_info& info,
+                          std::vector<instruction_ref> args) const
+    {
+        float bias = 0.0;
+        if(contains(info.attributes, "bias"))
+        {
+            bias = parser.parse_value(info.attributes.at("bias")).at<float>();
+        }
+        float lambd = 0.5;
+        if(contains(info.attributes, "lambd"))
+        {
+            lambd = parser.parse_value(info.attributes.at("lambd")).at<float>();
+        }
+
+        auto x_shape = args[0]->get_shape();
+        auto lit_bias =
+            info.add_literal(migraphx::literal{migraphx::shape{x_shape.type()}, {bias}});
+        auto lit_neg_lambd =
+            info.add_literal(migraphx::literal{migraphx::shape{x_shape.type()}, {-lambd}});
+        auto lit_lambd =
+            info.add_literal(migraphx::literal{migraphx::shape{x_shape.type()}, {lambd}});
+        auto lit_zero = info.add_literal(migraphx::literal{migraphx::shape{x_shape.type()}, {0}});
+
+        instruction_ref mb_bias;
+        instruction_ref mb_neg_lambd;
+        instruction_ref mb_lambd;
+        instruction_ref mb_zero;
+        if(x_shape.dynamic())
+        {
+            mb_bias = info.add_instruction(migraphx::make_op("multibroadcast"), lit_bias, args[0]);
+            mb_neg_lambd =
+                info.add_instruction(migraphx::make_op("multibroadcast"), lit_neg_lambd, args[0]);
+            mb_lambd =
+                info.add_instruction(migraphx::make_op("multibroadcast"), lit_lambd, args[0]);
+            mb_zero = info.add_instruction(migraphx::make_op("multibroadcast"), lit_zero, args[0]);
+        }
+        else
+        {
+            mb_bias = info.add_instruction(
+                migraphx::make_op("multibroadcast", {{"out_lens", x_shape.lens()}}), lit_bias);
+            mb_neg_lambd = info.add_instruction(
+                migraphx::make_op("multibroadcast", {{"out_lens", x_shape.lens()}}), lit_neg_lambd);
+            mb_lambd = info.add_instruction(
+                migraphx::make_op("multibroadcast", {{"out_lens", x_shape.lens()}}), lit_lambd);
+            mb_zero = info.add_instruction(
+                migraphx::make_op("multibroadcast", {{"out_lens", x_shape.lens()}}), lit_zero);
+        }
+
+        auto condition_1 = info.add_instruction(migraphx::make_op("less"), args[0], mb_neg_lambd);
+        auto condition_2 = info.add_instruction(migraphx::make_op("greater"), args[0], mb_lambd);
+
+        auto x_plus_bias = info.add_instruction(migraphx::make_op("add"), args[0], mb_bias);
+        auto x_min_bias  = info.add_instruction(migraphx::make_op("sub"), args[0], mb_bias);
+
+        auto filtered =
+            info.add_instruction(migraphx::make_op("where"), condition_1, x_plus_bias, mb_zero);
+        return info.add_instruction(migraphx::make_op("where"), condition_2, x_min_bias, filtered);
```

Comment:
ONNX spec is unclear to me if they want:
```
if(x < -lambd){ y = x + bias )
if(x > lambd){ y = x -bias }
else { y = 0}
```
or
```
if(x < -lambd){ y = x + bias )
else if(x > lambd){ y = x -bias }
else { y = 0}
```
It looks like this implements the `else if` version

Reply:
Good question for example `x=0` and `lamd=-5` can pass both conditions . I will check the reference implementation and come back with an answer.

Reply:
onnxruntime uses the `if`,`if`,`else` solution: https://github.com/microsoft/onnxruntime/blob/25bbd8d4ebed60c28f702a9fe6b0e6a12c19de9c/onnxruntime/core/providers/cpu/nn/shrink.cc#L37-L47